### DISCUSSION TITLE
Cannot connect to the server

### DISCUSSION THREAD

**Discussion by TrueReassembly** [Minestom/Minestom#1311] [Help] [answered]
``
class ServerJar {

    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
            val minecraftServer: MinecraftServer = MinecraftServer.init()
            val globalEventHandler: GlobalEventHandler = MinecraftServer.getGlobalEventHandler()
            val instanceManager = MinecraftServer.getInstanceManager()
            val instanceContainer = instanceManager.createInstanceContainer()

            MinecraftServer.getCommandManager().register(ServerStopCmd())
            instanceContainer.setGenerator {
                it.modifier().fillHeight(0, 40, Block.DIRT)
            }

            globalEventHandler.addListener(BasicQueryEvent::class.java) { event ->
                event.queryResponse.setMaxPlayers(100)
            }

            globalEventHandler.addListener(ServerListPingEvent::class.java) { event ->
                event.responseData.maxPlayer = 100;
                event.responseData.description = Component.text("Where pain is inescapable, pain is inescapable").color(NamedTextColor.DARK_RED)
            }

            globalEventHandler.addListener(PlayerLoginEvent::class.java) { event ->
                event.setSpawningInstance(instanceContainer)
                event.player.respawnPoint = Pos(0.0, 42.0, 0.0)
                event.player.skin = PlayerSkin.fromUsername(event.player.username)
            }

            globalEventHandler.addListener(PlayerBlockBreakEvent::class.java) { event ->
                if (!event.isCancelled) {
                    event.player.inventory.addItemStack(ItemStack.of(event.block.registry().material()!!))

                }
            }

            minecraftServer.start("0.0.0.0", 25575)
        }
    }
}``

On Ptero it shows as "starting" and the server cannot be connected to, even when minestom says the server is online

**Comment by DevBlokJochem** [comment]
Do you have an error?
Btw, if you want to use the server in online mode (so with skins etc), you can run this:
MojangAuth.init()

**Reply by TrueReassembly** [reply]
> Do you have an error? Btw, if you want to use the server in online mode (so with skins etc), you can run this: MojangAuth.init()

No errors, it just doesn't log in.

**Reply by TrueReassembly** [reply]
> > Do you have an error? Btw, if you want to use the server in online mode (so with skins etc), you can run this: MojangAuth.init()
> 
> No errors, it just doesn't log in.
`
container@pterodactyl~ Server marked as offline...
[Pterodactyl Daemon]: Updating process configuration files...
[Pterodactyl Daemon]: Ensuring file permissions are set correctly, this could take a few seconds...
container@pterodactyl~ Server marked as starting...
[Pterodactyl Daemon]: Pulling Docker container image, this could take a few minutes to complete...
Pulling from pterodactyl/yolks 
Status: Image is up to date for ghcr.io/pterodactyl/yolks:java_17 
Digest: sha256:46d8163f4493460caa607d65dbb665f76ed28add67594f8cd75bb5ad6fb4aa93 
[Pterodactyl Daemon]: Finished pulling Docker container image
container@pterodactyl~ java -version
openjdk version "17.0.2" 2022-01-18
OpenJDK Runtime Environment (build 17.0.2+8-86)
OpenJDK 64-Bit Server VM (build 17.0.2+8-86, mixed mode, sharing)
container@pterodactyl~ java -Xms128M -XX:MaxRAMPercentage=95.0 -jar server-1.0.jar
[main] [12:19:26] (ServerProcessImpl.start) - INFO - Starting Minestom server.
[main] [12:19:26] (ServerProcessImpl.start) - INFO - Minestom server started successfully.
`

**Comment by GreatWyrm** [comment]
This code seems to be just fine, and if nothing else is getting logged, then I would question if the port number is correct, since the default is 25565. Perhaps change that to 25565 and try connecting again.

**Comment by TrueReassembly** [comment, marked as answer]
I solved the issue, turns out I was using the wrong IP address, since the ptero panel gave me a different IP than what it actually was (node things)
